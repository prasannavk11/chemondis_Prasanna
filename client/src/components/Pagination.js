import React from "react";

const Pagination = (props) => {
  const { pageNumber, onChangePageNumber, disabled, noMoreData } = props;
  return (
    <nav aria-label="Page navigation example">
      <ul className="pagination justify-content-center">
        <li
          className={
            pageNumber === 0 || disabled
              ? "page-item disabled"
              : "page-item cursor-pointer"
          }
          onClick={onChangePageNumber}
        >
          <div className="page-link">Previous</div>
        </li>
        <li
          className={
            disabled || noMoreData
              ? "page-item disabled"
              : "page-item cursor-pointer"
          }
          onClick={onChangePageNumber}
        >
          <div className="page-link">Next</div>
        </li>
      </ul>
    </nav>
  );
};

export default Pagination;
