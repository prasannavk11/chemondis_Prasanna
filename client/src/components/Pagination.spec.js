import { render } from "@testing-library/react";
import Pagination from "./Pagination";

afterEach(() => {
  jest.clearAllMocks();
  jest.resetAllMocks();
});

describe("Pagination", () => {
  it("should render successfully", () => {
    const { baseElement } = render(
      <Pagination
        pageNumber={10}
        onChangePageNumber={() => {}}
        disabled={false}
        noMoreData={false}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
