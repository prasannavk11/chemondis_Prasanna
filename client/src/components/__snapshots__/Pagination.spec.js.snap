// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render successfully 1`] = `
<body>
  <div>
    <nav
      aria-label="Page navigation example"
    >
      <ul
        class="pagination justify-content-center"
      >
        <li
          class="page-item cursor-pointer"
        >
          <div
            class="page-link"
          >
            Previous
          </div>
        </li>
        <li
          class="page-item cursor-pointer"
        >
          <div
            class="page-link"
          >
            Next
          </div>
        </li>
      </ul>
    </nav>
  </div>
</body>
`;
